pandas/core/groupby/groupby.py

```diff
@@ -832,31 +843,37 @@ def _transform_should_cast(self, func_nm: str) -> bool:
         )
 
     def _cython_transform(self, how: str, numeric_only: bool = True, **kwargs):
-        output = collections.OrderedDict()  # type: dict
-        for obj in self._iterate_slices():
+        output: Dict[int, np.ndarray] = collections.OrderedDict()
+        names: List[Optional[Hashable]] = []
```

Comment:
why would this be Optional?

Reply:
None is a valid value but not covered by mypy in the Hashable umbrella so was throwing an error

Reply:
The `Optional[Hashable]` thing has bugged me, too.

Why is output `Dict` here instead of `Mapping` like most other places?

Reply:
You cannot assign to a Mapping; we typically use Mapping for our interface as it is more generic when accepting a dict as an argument, but can't be used internally when actually assigning values to an object